lib/_emerge/depgraph.py

```diff
@@ -1288,17 +1288,33 @@ def _show_ignored_binaries_respect_use(self, respect_use):
             writemsg(line + "\n", noiselevel=-1)
 
     def _show_ignored_binaries_changed_deps(self, changed_deps):
-        writemsg(
-            "\n!!! The following binary packages have been "
-            "ignored due to changed dependencies:\n\n",
-            noiselevel=-1,
-        )
+        merging = {
+            (pkg.root, pkg.cpv)
+            for pkg in self._dynamic_config._displayed_list or ()
+            if isinstance(pkg, Package)
+        }
+        messages = []
 
         for pkg in changed_deps:
+            # Don't include recursive deps which aren't in the merge list anyway.
+            if (pkg.root, pkg.cpv) not in merging:
+                continue
+
```

Comment:
It doesn't really matter to me since I'm not really interested in these messages, but we could conceivable use a map of packages indexed by `(pkg.root, pkg.cp, pkg.slot)` here, and display the message only when the ignored package version is greater than or equal to a selected package with the same `(pkg.root, pkg.cp, pkg.slot)`.